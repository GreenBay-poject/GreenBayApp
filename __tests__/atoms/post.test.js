import React from 'react';
import renderer from 'react-test-renderer';

import PostCard from '../../atoms/post'

describe('<PostCard />', () => {
    it('Sample PostCard With no Null Image', () => {
        const tree = renderer.create(<PostCard Image={"https://www.testbytes.net/wp-content/uploads/2019/12/benefits-of-end-to-end-testing-1-scaled.jpg"} Title={"Title 1"} Description={"Sample Desc."} DatePosted={"2021-08-02"} email={"SamaSilva@gmail.com"} post_id={"24240"} />).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree.children[0].children.length).toBe(8);
    });
    it('Sample PostCard With Null Image', () => {
        const tree = renderer.create(<PostCard Image={""} Title={"Title 1"} Description={"Sample Desc."} DatePosted={"2021-08-02"} email={"SamaSilva@gmail.com"} post_id={"24240"} />).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree.children[0].children.length).toBe(8);
    });
});