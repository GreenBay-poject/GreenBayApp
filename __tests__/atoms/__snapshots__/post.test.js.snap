// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> Sample PostCard With Null Image 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 0,
      "padding": 15,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#43484d",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 15,
          "textAlign": "center",
        }
      }
      testID="cardTitle"
    >
      Title 1
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#bcbbc1",
          "borderBottomWidth": 0.5,
          "marginBottom": 15,
        }
      }
    />
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <Image
        onLoad={[Function]}
        source={
          Object {
            "uri": "https://cdn2.wanderlust.co.uk/media/1037/forest-web.jpg?anchor=center&mode=crop&width=1200&height=0&rnd=132605629110000000",
          }
        }
        style={
          Object {
            "bottom": 0,
            "height": 150,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": null,
          }
        }
        testID="RNE__Image"
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
        transition={true}
        transitionDuration={360}
      />
      <View
        accessibilityElementsHidden={true}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#bdbdbd",
              "height": 150,
              "justifyContent": "center",
              "width": null,
            }
          }
          testID="RNE__Image__placeholder"
        />
      </View>
      <View
        style={
          Object {
            "height": 150,
            "width": null,
          }
        }
        testID="RNE__Image__children__container"
      />
    </View>
    <Text
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      Sample Desc.
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#bcbbc1",
          "borderBottomWidth": 0.5,
          "marginBottom": 15,
          "marginTop": 10,
        }
      }
    />
    <Text>
      Posted By : SamaSilva@gmail.com
    </Text>
    <Text
      style={Object {}}
    >
      Post ID : 24240
    </Text>
    <Text
      style={Object {}}
    >
      Date : 2021-08-02
    </Text>
  </View>
</View>
`;

exports[`<PostCard /> Sample PostCard With no Null Image 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 0,
      "padding": 15,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#43484d",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 15,
          "textAlign": "center",
        }
      }
      testID="cardTitle"
    >
      Title 1
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#bcbbc1",
          "borderBottomWidth": 0.5,
          "marginBottom": 15,
        }
      }
    />
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <Image
        onLoad={[Function]}
        source={
          Object {
            "uri": "https://www.testbytes.net/wp-content/uploads/2019/12/benefits-of-end-to-end-testing-1-scaled.jpg",
          }
        }
        style={
          Object {
            "bottom": 0,
            "height": 150,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": null,
          }
        }
        testID="RNE__Image"
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
        transition={true}
        transitionDuration={360}
      />
      <View
        accessibilityElementsHidden={true}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#bdbdbd",
              "height": 150,
              "justifyContent": "center",
              "width": null,
            }
          }
          testID="RNE__Image__placeholder"
        />
      </View>
      <View
        style={
          Object {
            "height": 150,
            "width": null,
          }
        }
        testID="RNE__Image__children__container"
      />
    </View>
    <Text
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      Sample Desc.
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#bcbbc1",
          "borderBottomWidth": 0.5,
          "marginBottom": 15,
          "marginTop": 10,
        }
      }
    />
    <Text>
      Posted By : SamaSilva@gmail.com
    </Text>
    <Text
      style={Object {}}
    >
      Post ID : 24240
    </Text>
    <Text
      style={Object {}}
    >
      Date : 2021-08-02
    </Text>
  </View>
</View>
`;
